src/libs/actions/Task.js

```diff
@@ -335,10 +364,32 @@ function reopenTask(taskReportID, taskTitle) {
         },
     ];
 
-    // Update optimistic data for parent report action
+    // Update optimistic data and failure data for parent report action
     const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, reopenedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
     if (!_.isEmpty(optimisticParentReportData)) {
         optimisticData.push(optimisticParentReportData);
+        const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants(taskReport.participantAccountIDs), 'reportID');
+        if (assigneeChatReportID && `${assigneeChatReportID}` !== `${taskReport.parentReportID}`) {
+            const parentReportAction = ReportActionsUtils.getParentReportActionForTask(taskReportID, assigneeChatReportID);
+            if (parentReportAction) {
+                optimisticData.push({
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+                    value: {
+                        [parentReportAction.reportActionID]: _.values(optimisticParentReportData.value)[0],
+                    },
+                });
+                failureData.push({
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+                    value: {
+                        [parentReportAction.reportActionID]: {
+                            errors: ErrorUtils.getMicroSecondOnyxError('task.messages.error'),
+                        },
+                    },
+                });
```

Comment:
Do we need failure data hare? Won't it cause a double redundant error messages? or in stuck errors? Asking because `optimisticParentReportData` for the share report does not have a failure data

Reply:
Let me remove it for consistency

Reply:
removed